integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java

```diff
@@ -471,15 +480,27 @@ private int releaseInternal(String path, FuseFileInfo fi) {
         LOG.error("Cannot find fd {} for {}", fd, path);
         return -ErrorCodes.EBADFD();
       }
-      if (is != null) {
-        synchronized (is) {
-          is.close();
-        }
-      }
       if (ce != null) {
+        // Remove earlier to try best effort to avoid write() - async release() - getAttr()
+        // without waiting for file completed and return 0 bytes file size error
         mCreateFileEntries.remove(ce);
-        synchronized (ce) {
-          ce.getOut().close();
+        mReleasingWriteEntries.put(fd, ce);
```

Comment:
Seems the `fd` and `ce` will be kept in mReleasingWriteEntries until we umount the fuse daemon. Should we check and clean  the `mReleasingWriteEntries` periodically? Or move the logic of "put ce into mReleasingWriteEntries" in `create` function and remove it after close is done in server side

Reply:
They will only be kept in `mReleasingWriteEntries` when they are releasing. It should be a short period. No need to clean it periodically.

Reply:
For this change, I think `mReleasingWriteEntries` will only be cleaned when `umount` function is called.
In some deployment, fuse daemon will run as a service. So umount will no be called until admin stop the service. 
For this case, all closed file entries will be kept in `mReleasingWriteEntries`. If we read/write more than 100 million files (which is the common  for imageNet dataset),  then  `mReleasingWriteEntries` will consume much memory resource.

Reply:
@Binyang2014 ` mReleasingWriteEntries.remove(fd);` in line 495 after `ce.close()` completes

Reply:
Got it thanks. I miss this piece